pkg/controllers/dnspolicy/dnspolicy_dnsrecords.go

```diff
@@ -135,3 +132,13 @@ func (r *DNSPolicyReconciler) deleteGatewayDNSRecords(ctx context.Context, gatew
 	}
 	return nil
 }
+
+func getPlacedClusters(gateway *gatewayv1beta1.Gateway) []string {
+	var clusters []string
+
+	for _, address := range gateway.Status.Addresses {
+		clusters = append(clusters, strings.Split(address.Value, "/")[0])
+	}
+
+	return clusters
```

Comment:
Right now it will ignore ISTO gateways (`clusters` will be empty) and then it will see that we have no `clusterGateways` for this gateway and will opt out of the reconciler cleaning up remaining DNSRecords for this gateway.